### ISSUE TITLE
ODBC string bulk insertions are adding an escape char at the end of each string

### ISSUE THREAD

**Issue by hochlior** [SOCI/soci#202]
Hi
we are using your great library and found out that if we are inserting via std::vector of std::string (bulk insertion), then for each string a char(0) is added at the end of the string.
For example: 

``` cpp
std::vector<std::string> names;
names.push_back("Simple");
names.push_back("Simple test a");
sql << "INSERT INTO myTest (Name) VALUES (:n)", use(names);
```

This does not happens in regular string insertion.

**Comment by candychiu** [comment]
@mloskot @vadz 
I verified the problem exists and proposes to fix it by removing the "+1" in vector-use-type.cpp line 168.  I haven't been able to set up and run unit tests on my development environment,  would you guys try it out and if it works, check in this fix?

**Comment by vadz** [comment]
If you submit a PR with the proposed fix, Travis CI will automatically run unit tests with many databases.

From looking at the code, it seems that removing `+1` is indeed the right fix, at least assuming the strings don't need to be NUL-terminated (because they won't be).

I also think that using `strncpy()` in this code is quite useless, we should be just using `memcpy()` for (small) efficiency (gain) and, more importantly, for clarity.

**Comment by vadz** [comment]
This will soon be (finally!) fixed by #561.
